File: backend/test-qdrant.ts
import { config } from "dotenv";
config(); // Load .env file

console.log("QDRANT_URL:", process.env.QDRANT_URL);
console.log("QDRANT_API_KEY exists:", !!process.env.QDRANT_API_KEY);

import { QdrantClient } from "@qdrant/js-client-rest";

// Initialize Qdrant client with logging
console.log("Creating Qdrant client with URL:", process.env.QDRANT_URL);

const qdrant = new QdrantClient({
  url: process.env.QDRANT_URL!,
  apiKey: process.env.QDRANT_API_KEY!,
  checkVersionCompatibility: false, // Skip version compatibility check
});

console.log("Qdrant client created. Testing connection...");

async function testConnection() {
  try {
    // Try to get collection info to test the connection
    const info = await qdrant.getCollections();
    console.log("Connection successful! Collections:", info.collections.length);

    // Check if our collection exists
    try {
      const collectionInfo = await qdrant.getCollection("book_chunks_local");
      console.log("Collection info:", collectionInfo);
    } catch (err) {
      console.log("Collection 'book_chunks_local' doesn't exist yet:", err.message);
    }
  } catch (error) {
    console.error("Connection failed:", error.message);
    console.error("Error cause:", error.cause);
  }
}

testConnection();

File: backend/test_qdrant.py
import asyncio
from qdrant_client import AsyncQdrantClient
import os
from dotenv import load_dotenv

load_dotenv()

async def test_qdrant():
    url = os.getenv("QDRANT_URL")
    if not url.endswith(":6333"):
        url = url.rstrip("/") + ":6333"
    api_key = os.getenv("QDRANT_API_KEY")
    print(f"Testing Qdrant at: {url}")
    client = AsyncQdrantClient(url=url, api_key=api_key)
    try:
        collections = await client.get_collections()
        print(f"Successfully connected! Collections: {collections}")
    except Exception as e:
        print(f"Failed to connect: {e}")

if __name__ == "__main__":
    asyncio.run(test_qdrant())
